Dockerfile

```diff
@@ -1,11 +1,23 @@
-FROM node:17-bullseye-slim
+FROM node:17-bullseye-slim AS chrono-build-stage
 ENV NODE_ENV production
-RUN apt-get update && apt-get install -y --no-install-recommends dumb-init
+WORKDIR /usr/src/app
+
+COPY client .
+
+RUN npm ci --omit=dev && NODE_OPTIONS=--openssl-legacy-provider SKIP_PREFLIGHT_CHECK=true npm run build
+
+# Deploy stage
+FROM node:17-bullseye-slim
 WORKDIR /usr/src/app
 
 COPY --chown=node:node . .
+RUN rm -rf client
+RUN mkdir -p client/build
+COPY --from=chrono-build-stage --chown=node:node /usr/src/app/build /usr/src/app/client/build
+
+RUN apt-get update && apt-get install -y --no-install-recommends dumb-init && apt-get clean && rm -rf /var/lib/apt/lists/*
```

Comment:
I think explicitly setting `NODE_ENV` again in the deploy stage would be a good idea. Everything else looks good.

Reply:
I'll do this, and I'll be optimizing it a little more (docker cache works really well with pnpm)